simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/HeartbeatScheduler.java

```diff
@@ -4,25 +4,24 @@
 import lombok.extern.slf4j.Slf4j;
 
 import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 
 @Slf4j
 @FieldDefaults(makeFinal = true)
 public class HeartbeatScheduler {
 
-    private static final int INITIAL_TASK_DELAY_IN_SECONDS = 1;
-    private static final int TASK_DELAY_IN_SECONDS = 1;
-
     private HeartbeatSenderTask heartbeatTask;
+    private ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
 
     public HeartbeatScheduler(StationState stationState, StationMessageSender stationMessageSender) {
         this.heartbeatTask = new HeartbeatSenderTask(stationState, stationMessageSender);
-        Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(
-                heartbeatTask, INITIAL_TASK_DELAY_IN_SECONDS, TASK_DELAY_IN_SECONDS, TimeUnit.SECONDS);
     }
 
     public void updateHeartbeat(int heartbeatInterval) {
         log.debug("Updating heartbeat to {} sec.", heartbeatInterval);
         heartbeatTask.updateHeartBeatInterval(heartbeatInterval);
+        scheduledExecutorService.scheduleAtFixedRate(
```

Comment:
This is not correct behavior. This task should run every second, not every heartbeatInterval.

Reply:
Right. Changed it back